airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -21,70 +21,222 @@
+        """
+        Return the latest state by comparing the campaign_id and cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+        current_state = current_stream_state.get(self.campaign_id) if current_stream_state else None
+        latest_state = latest_record.get(self.campaign_id) if latest_record else None
+        if not any([current_state, latest_state]):  # if no state was passed
+            # constuct dict state first time
+            current_state = current_stream_state.get(self.cursor_field)
+            latest_state = latest_record.get(self.cursor_field)
+            if not all([current_state, latest_state]):
+                max_value = list(filter(bool, [current_state, latest_state]))[0]
+            else:
+                max_value = max(latest_state, current_state)
+            output = {self.campaign_id: {self.cursor_field: max_value}}
+            return output
+
+        current_campaign_state = current_state.get(self.cursor_field) if current_state else None
+        latest_campaign_state = latest_state.get(self.cursor_field) if latest_state else None
+        if not all([current_campaign_state, latest_campaign_state]):
+            nonzero_state = list(filter(bool, [current_campaign_state, latest_campaign_state]))[0]
+            output = {self.campaign_id: {self.cursor_field: nonzero_state}}
+        else:
+            max_value = max(current_campaign_state, latest_campaign_state)
+            output = {self.campaign_id: {self.cursor_field: max_value}}
+        return output
+
+    def request_params(self, stream_state=None, **kwargs):
+        stream_state = stream_state or {}
+        params = MailChimpStream.request_params(self, stream_state=stream_state, **kwargs)
+
+        since_value_camp = stream_state.get(self.campaign_id)
+        if since_value_camp:
+            since_value = since_value_camp.get(self.cursor_field)
+            if since_value:
+                params["since"] = since_value
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        # transform before save
+        # [{'campaign_id', 'list_id', 'list_is_active', 'email_id', 'email_address', 'activity[array[object]]', '_links'}] ->
+        # -> [[{'campaign_id', 'list_id', 'list_is_active', 'email_id', 'email_address', '**activity[i]', '_links'}, ...]]
+        data = response_json[self.data_field]
+        for item in data:
+            for activity_record in item["activity"]:
+                new_record = {k: v for k, v in item.items() if k != "activity"}
+                for k, v in activity_record.items():
+                    new_record[k] = v
+                yield new_record
+
+
+class HttpBasicAuthenticator(HttpAuthenticator):  # should not be placed in this file but we havent it!
```

Comment:
it is cleaner to inherit from TokenAuthenticator,
set auth_method="Basic", and token to b64_encoded
then you don't need `get_auth_header`